File: src/main/java/de/n2online/sonification/Keyboard.java
package de.n2online.sonification;

import javafx.scene.input.KeyCode;
import javafx.scene.input.KeyEvent;

import java.util.HashSet;

public class Keyboard {
    private HashSet<KeyCode> activeCodes;

    public Keyboard() {
        activeCodes = new HashSet<>();
    }

    public void registerKeyDown(KeyEvent event) {
        KeyCode code = event.getCode();
        activeCodes.add(code);
    }

    public void registerKeyUp(KeyEvent event) {
        KeyCode code = event.getCode();
        activeCodes.remove(code);
    }

    public boolean isKeyDown(KeyCode code) {
        return activeCodes.contains(code);
    }
}

File: src/main/scala/de/n2online/sonification/Keyboard.scala
package de.n2online.sonification

import javafx.scene.input.{KeyCode, KeyEvent}

class Keyboard {
  private var activeCodes: Set[KeyCode] = Set()
  var consumeEvents = false

  def registerKeyDown(event: KeyEvent) {
    activeCodes = activeCodes + event.getCode
    if (consumeEvents) event.consume()
  }

  def registerKeyUp(event: KeyEvent) {
    activeCodes = activeCodes - event.getCode
    if (consumeEvents) event.consume()
  }

  def isKeyDown(code: KeyCode): Boolean = {
    activeCodes(code)
  }
}